plugin/lighthouse/handler.go

```diff
@@ -0,0 +1,63 @@
+package lighthouse
+
+import (
+	"context"
+	"net"
+	"strings"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+)
+
+// ServeDNS implements the plugin.Handler interface.
+func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	state := request.Request{W: w, Req: r}
+	// Debug log that we've have seen the query. This will only be shown when the debug plugin is loaded.
+
+	a := new(dns.Msg)
+	a.SetReply(r)
+	a.Authoritative = true
+
+	services := lh.SvcsMap
+	svcName := strings.Split(state.QName(), ".")[0]
+	ip := services[svcName]
+	if ip == "" {
+		return onFailure(lh, ctx, w, r)
+	}
+	var rr dns.RR
```

Comment:
no need to do this, as only case IPv4 is being implemented here. Thus you could just put `:=` on L34

Reply:
Done. scope of variable declared in switch/case is within switch case, so declared it here.